src/main/java/org/stellar/sdk/Sep10Challenge.java

```diff
@@ -0,0 +1,39 @@
+package org.stellar.sdk;
+
+import java.security.SecureRandom;
+
+public class Sep10Challenge {
+  /**
+   * Returns a valid <a href="https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md#response" target="_blank">SEP 10</a> challenge, for use in web authentication.
+   * @param signer           The server's signing account.
+   * @param network          The Stellar network used by the server.
+   * @param clientAccountId  The stellar account belonging to the client.
+   * @param anchorName       The name of the anchor which will be included in the ManageData operation.
+   * @param timebounds       The lifetime of the challenge token.
+   */
+  public static String newChallenge(
+      KeyPair signer,
+      Network network,
+      String clientAccountId,
+      String anchorName,
+      TimeBounds timebounds
+  ) {
+    byte[] nonce = new byte[64];
+    SecureRandom random = new SecureRandom();
+    random.nextBytes(nonce);
```

Comment:
You may want to check a discussion here: https://github.com/stellar/stellar-protocol/pull/366#discussion_r330134927. I don't fully agree with reasoning there but it looks like you need to generate 48 bytes and base64 encode them to fully comply with SEP-10.